src/visualization/utilities.jl

```diff
@@ -766,31 +766,38 @@ end
 
 # Calculate the arc length of a curve given by ndims x npoints point coordinates (piece-wise linear approximation)
 function calc_arc_length(coordinates)
-    n_points = size(coordinates)[2]
+    n_points = size(coordinates, 2)
     arc_length = zeros(n_points)
     for i in 1:(n_points - 1)
-        arc_length[i + 1] = arc_length[i] +
-                            sqrt(sum((coordinates[:, i] - coordinates[:, i + 1]) .^ 2))
+        res = zero(eltype(arc_length))
+        for j in axes(coordinates, 1)
+            res += (coordinates[j, i + 1] - coordinates[j, i])^2
+        end
+        arc_length[i + 1] = arc_length[i] + sqrt(res)
     end
     return arc_length
 end
 
 # Convert 2d unstructured data to 1d data at given curve.
 function unstructured_2d_to_1d_curve(original_nodes, unstructured_data, nvisnodes,
                                      curve, mesh, solver, cache)
+    # Extract points as `SVector`s
+    get_point(data, idx...) = SVector(data[1, idx...], data[2, idx...])
+
```

Comment:
Could we move a function definition like this to right before it's called? It took me a second to realize where `get_point` was defined.